packages/embark-ui/src/containers/ContractsContainer.js

```diff
@@ -23,14 +29,58 @@ class ContractsContainer extends Component {
     this.props.stopContracts();
   }
 
+  get numberOfContracts() {
+    return this.props.contracts
+      .filter(contract => !contract.silent).length;
+  }
+
+  getNumberOfPages() {
+    return Math.ceil(this.numberOfContracts / this.numContractsToDisplay);
+  }
+
+  changePage(newPage) {
+    this.setState({currentPage: newPage});
```

Comment:
Should we validate that it's a valid page? In case the arrow somehow enabled going to -1

Reply:
How does this sound:

if the page number `<= 0` or if the page number `> getNumberOfPages()` then set the page to `1`.

Reply:
I'd say, if `<=0`, then put 0, if `> getNumberOfPages()`, then set `getNumberOfPages()`

Reply:
Ah yeah that makes sense re: `getNumberOfPages`. However, the first page is `1` not `0` that would the reason for setting to `1` if it's `<= 0`.

Reply:
Done. See WIP commit.